pkg/log/null.go

```diff
@@ -25,23 +29,46 @@ import (
 // all binaries.
 
 // NullLogger is a logr.Logger that does nothing.
-type NullLogger struct{}
+type NullLogger struct {
```

Comment:
I don't think this makes sense. The null logger shouldn't consider itself to be problematic. I am fine with the warning when it is being set in the delgating logger, but not here